smalldata_tools/SmallDataDefaultDetector.py

```diff
@@ -680,27 +682,43 @@ def data(self,evt):
         try:
             self._XTCAVRetrieval.processEvent(evt)
             dl['success']=1
-            t, power = self._XTCAVRetrieval.xRayPower()
-            dl['success']=2
-            if t is not None:
-                imethod=0
-                for this_t, this_power in zip(t, power):
-                    arSize=this_t.shape[0]
-                    if arSize>=1 and arSize<=self.size:
-                        timeAr = np.append(this_t,np.array([np.nan] * (self.size-arSize)))
-                        powerAr = np.append(this_power,np.array([np.nan] * (self.size-arSize)))
-                    else:
-                        print('Xtcav array is too small in run, please check configuration',env.run())
-                        timeAr = this_t[:self.size]
-                        powerAr = this_power[:self.size]
-                    #we may have to create this outside for 
-                    #  data with much damage, 
-                    #  assuming we only have 1 method
-                    dl['arSize_%d'%imethod]=arSize
-                    dl['time_%d'%imethod]=timeAr
-                    dl['power_%d'%imethod]=powerAr
-                    imethod+=1
-                dl['success']=3
+            if self.method=='COM' or self.method=='RMS':
+                t, power = self._XTCAVRetrieval.xRayPower(self.method)
+                dl['success']=2
+                if t is not None:
+                    imethod=0
+                    for this_t, this_power in zip(t, power):
+                        arSize=this_t.shape[0]
+                        if arSize>=1 and arSize<=self.size:
+                            timeAr = np.append(this_t,np.array([np.nan] * (self.size-arSize)))
+                            powerAr = np.append(this_power,np.array([np.nan] * (self.size-arSize)))
+                        else:
+                            print('Xtcav array is too small in run, please check configuration',env.run())
+                            timeAr = this_t[:self.size]
+                            powerAr = this_power[:self.size]
+                        #we may have to create this outside for 
+                        #  data with much damage, 
+                        #  assuming we only have 1 method
+                        dl['arSize_%d'%imethod]=arSize
+                        dl['time_%d'%imethod]=timeAr
+                        dl['power_%d'%imethod]=powerAr
+                        imethod+=1
+                    dl['success']=3
+            else:
+                full_results = self._XTCAVRetrieval.fullResults()
+                keylist=['grounpnum','num_bunches','bunchdelay','nolasingECOM','lasingECOM']
```

Comment:
Never used, what is this for?

Reply:
Debug leftovers. I'll try to remember to take them out another time...